avogadro/qtplugins/bonding/bonding.cpp

```diff
@@ -95,8 +98,14 @@ void Bonding::bond2()
 
   // Main bond perception loop based on a simple distance metric.
   for (Index i = 0; i < m_molecule->atomCount(); ++i) {
+    if (!m_molecule->atomSelected(i))
```

Comment:
This means I will only get any bonding if all atoms are selected? It seems like if none are selected you might just prefer global bond perception (rather than nothing happening and assuming it is broken). Maybe storing if selection is empty, and only skipping if not empty and the atom is selected would make more sense.